core/run/src/lib.rs

```
@@ -145,6 +147,15 @@ async fn start<K: KeyProvider>(
     let current_block = storage.get_latest_block(Context::new()).await?;
     let current_state_root = current_block.header.state_root;
 
+    if current_block.header.number
```

I think it should throw an error to tell users that your setting is invalid when hardfork number is less than current number.
So, users could change hardfork number a.s.a.p.

---

+1